Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +353,278 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+        ( 1.,  2., 0, 0.00397386), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf), (inf, inf, 0,        inf),
+        (inf, inf, 0,        inf)],
+       dtype=[('nRA', '<f8'), ('nDEC', '<f8'), ('avg_time', '<u8'), \
+       ('sigma_time', '<f8')]))
+    """
+
+    RA1 = RA
+    DEC1 = DEC
+    RA2 = None
+    DEC2 = None
+
+    weights1 = None
+    weights2 = None
+    weight_type = None
+
+    try:
+        from Corrfunc._countpairs_mocks import countpairs_theta_mocks as \
+            DDtheta_mocks_extn
+    except ImportError:
+        msg = "Could not import the C extension for the angular " \
+              "correlation function for mocks."
+        raise ImportError(msg)
+
+    import numpy as np
+    from Corrfunc.utils import translate_isa_string_to_enum, fix_ra_dec, \
+        return_file_with_rbins, convert_to_native_endian, process_weights
+    from future.utils import bytes_to_native_str
+    import itertools
+    import time
+
+    if autocorr == 0:
+        if RA2 is None or DEC2 is None:
+            msg = "Must pass valid arrays for RA2/DEC2 for " \
+                  "computing cross-correlation"
+            raise ValueError(msg)
+    else:
+        RA2 = np.empty(1)
+        DEC2 = np.empty(1)
+
+    weights1, weights2 = \
+        process_weights(weights1, weights2, RA1, RA2, weight_type, autocorr)
+
+    # Ensure all input arrays are native endian
+    RA1, DEC1, weights1, RA2, DEC2, weights2 = [
+        convert_to_native_endian(arr, warn=True) for arr in
+        [RA1, DEC1, weights1, RA2, DEC2, weights2]]
+
+    fix_ra_dec(RA1, DEC1)
+    if autocorr == 0:
+        fix_ra_dec(RA2, DEC2)
+
+    # Passing None parameters breaks the parsing code, so avoid this
+    kwargs = {}
+    for k in ['weights1', 'weights2', 'weight_type', 'RA2', 'DEC2']:
+        v = locals()[k]
+        if v is not None:
+            kwargs[k] = v
+
+    integer_isa = translate_isa_string_to_enum(isa)
+    rbinfile, delete_after_use = return_file_with_rbins(binfile)
+    bin_refs = np.arange(1, maxbinref + 1)
+    bin_ref_perms = itertools.product(bin_refs, bin_refs)
+    dtype = np.dtype([(bytes_to_native_str(b'nRA'), np.float),
+                      (bytes_to_native_str(b'nDEC'), np.float),
+                      (bytes_to_native_str(b'avg_time'), np.uint64),
+                      (bytes_to_native_str(b'sigma_time'), np.float)])
+    all_runtimes = np.zeros(maxbinref ** 3, dtype=dtype)
+    all_runtimes[:] = np.inf
+
+    for ii, (nRA, nDEC) in enumerate(bin_ref_perms):
```

Comment:
`DDtheta_mocks` has potentially three different runtime modes:

- `(link_in_dec=False, link_In_ra=False)` -> brute-force calculation without any gridding
- `(link_in_dec=True, link_in_ra=False)`  -> only gridding in declination. Only need to loop over range of (min, max) bin-refinements in declination
- `(link_in_dec=True, link_in_ra=True)` -> gridding in both RA and DEC. However,the declination bin-loop must preceede that on the ra-bin-loop. The current implementation as written covers this case

Have a think about how to handle the first two cases and return an appropriate result (or info message) back to the user

Reply:
Sure, I have used `raise ValueError(msg#)` to handle the first two cases and return some proper messages back to the user if those two are the case. Please let me know if any further changes is necessary.

Reply:
You should still be able to loop over a range of `dec-bin-refine factors` for the second case. For the first one, if you are raising a ValueError, then please make sure that the error message conveys **why** the error was raised **and** how to fix the error (i.e., turn on `link_in_dec, link_in_ra`)

Reply:
Thanks for the feedback! 

For the "brute-force" Case 1, where `(link_in_dec=False, link_In_ra=False)`, I am now raising an error with the following message instead: 
> "Warning: Brute-force calculation without any gridding is forced, as link_in_dec and link_in_ra are both set to False. Please be sure to turn on at least link_in_dec, or both link_in_dec and link_in_ra on."

For Case 2 where `(link_in_dec=True, link_in_ra=False)`, I am now switching to use a print statement instead with the following message:
> "Note: Gridding in the declination only, as link_in_dec is set to True while link_in_ra is set to False. Thus looping is only needed over the range of (min, max) bin, with refinements in the declination."